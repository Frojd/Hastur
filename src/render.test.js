const {
    getComponentElement,
    renderServerComponent,
    renderStaticServerComponent,
} = require('./render');

describe('Render a component', function() {
    it('should return a react element', () => {
        const element = getComponentElement(`${process.cwd()}/test/ReactComponent`, {});
        expect(element).toMatchSnapshot();
    })

    it('should be able to return a html representation of a reactcomponent', function() {
        const component = renderServerComponent(`${process.cwd()}/test/ReactComponent`, {});
        expect(component).toMatchSnapshot();
    });

    it('should be able to return a static html representation of a reactcomponent', function() {
        const component = renderStaticServerComponent(`${process.cwd()}/test/ReactComponent`, {});
        expect(component).toMatchSnapshot();
    });

    it('should be able to return a html representation of a reactcomponent with props', function() {
        const props = {
            "title": "Hello World!"
        }
        const component = renderServerComponent(`${process.cwd()}/test/ReactComponent`, props);
        expect(component).toMatchSnapshot();
    });
});