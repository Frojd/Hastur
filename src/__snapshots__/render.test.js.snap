// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a component should be able to return a html representation of a reactcomponent 1`] = `"<div class=\\"ReactComponent\\" data-reactroot=\\"\\">ReactComponent </div>"`;

exports[`Render a component should be able to return a html representation of a reactcomponent with props 1`] = `"<div class=\\"ReactComponent\\" data-reactroot=\\"\\">ReactComponent <!-- -->Hello World!</div>"`;

exports[`Render a component should be able to return a static html representation of a reactcomponent 1`] = `"<div class=\\"ReactComponent\\">ReactComponent </div>"`;

exports[`Render a component should return a react element 1`] = `<ReactComponent />`;
